extensions/amp-sidebar/1.0/amp-sidebar.js

```diff
@@ -203,33 +212,42 @@ export class AmpSidebar extends AMP.BaseElement {
     element.addEventListener(
       'click',
       e => {
-        const target = closestAncestorElementBySelector(
-          dev().assertElement(e.target),
-          'A'
-        );
-        if (target && target.href) {
-          const tgtLoc = Services.urlForDoc(element).parse(target.href);
-          const currentHref = this.getAmpDoc().win.location.href;
-          // Important: Only close sidebar (and hence pop sidebar history entry)
-          // when navigating locally, Chrome might cancel navigation request
-          // due to after-navigation history manipulation inside a timer callback.
-          // See this issue for more details:
-          // https://github.com/ampproject/amphtml/issues/6585
-          if (removeFragment(target.href) != removeFragment(currentHref)) {
-            return;
-          }
-
-          if (tgtLoc.hash) {
-            this.close_();
-          }
-        }
+        this.handleAnchorClick_(e);
+        this.submenuManager_.handleClick(e);
```

Comment:
Looks like only one of these two would handle the click, right? If so I'd make these return a boolean based on whether they handled the click or not and only handle the click on the submenu level if the anchor returns false. Same for the two methods inside `this.submenuManager_ .handleClick`.

Reply:
Cathy recommended adding event listeners to each open/close element independently rather than using event delegation, in which case we'll no longer need the conditionals.